pkg/network/protocols/http2/model_linux.go

```diff
@@ -147,38 +147,39 @@ func stringToHTTPMethod(method string) (http.Method, error) {
 		return http.MethodOptions, nil
 	case "PATCH":
 		return http.MethodPatch, nil
-	// Add more cases for other HTTP methods as needed
+	case "GET":
+		return http.MethodGet, nil
+	case "POST":
+		return http.MethodPost, nil
+	// Currently unsupported methods due to lack of support in http.Method.
+	case "CONNECT":
+		return http.MethodUnknown, nil
+	case "TRACE":
+		return http.MethodUnknown, nil
 	default:
 		return 0, fmt.Errorf("unsupported HTTP method: %s", method)
 	}
 }
 
 // Method returns the HTTP method of the transaction.
 func (tx *EbpfTx) Method() http.Method {
-	if tx.Stream.Request_method.Static_table_entry != 0 {
+	var method string
+	var err error
+	if tx.Stream.Request_method.Is_huffman_encoded {
+		method, err = hpack.HuffmanDecodeToString(tx.Stream.Request_method.Raw_buffer[:tx.Stream.Request_method.Length])
+		if err != nil {
+			return 0
+		}
+	} else {
 		switch tx.Stream.Request_method.Static_table_entry {
 		case GetValue:
 			return http.MethodGet
 		case PostValue:
 			return http.MethodPost
 		default:
-			return http.MethodUnknown
-		}
-	}
-
-	if tx.Stream.Request_method.Is_huffman_encoded {
-		method, err := hpack.HuffmanDecodeToString(tx.Stream.Request_method.Raw_buffer[:tx.Stream.Request_method.Length])
-		if err != nil {
-			return 0
+			method = string(tx.Stream.Request_method.Raw_buffer[:tx.Stream.Request_method.Length])
```

Comment:
the new format does not make sense

Reply:
you're handling the "literal but not encoded" scenario inside the "static table" use-cases

Reply:
you are right, changed